src/adios4dolfinx/checkpointing.py

```diff
@@ -296,7 +296,7 @@ def read_meshtags(filename: str, mesh: dolfinx.mesh.Mesh, meshtag_name: str,
 
     # Memory leak due to nanobind, ref: https://github.com/FEniCS/dolfinx/issues/2997
     local_entities, local_values = dolfinx.cpp.io.distribute_entity_data(
-         mesh._cpp_object, int(dim), mesh_entities, tag_values)
+        mesh._cpp_object, int(dim), mesh_entities, tag_values)
```

Comment:
Looks like this issue is now resolved https://github.com/FEniCS/dolfinx/issues/2997